compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2438,17 +2438,6 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             | ObligationCauseCode::ExprBindingObligation(item_def_id, span, ..) => {
                 let item_name = tcx.def_path_str(item_def_id);
                 let mut multispan = MultiSpan::from(span);
-                if let Some(ident) = tcx.opt_item_ident(item_def_id) {
-                    let sm = tcx.sess.source_map();
-                    let same_line =
-                        match (sm.lookup_line(ident.span.hi()), sm.lookup_line(span.lo())) {
-                            (Ok(l), Ok(r)) => l.line == r.line,
-                            _ => true,
-                        };
-                    if !ident.span.overlaps(span) && !same_line {
-                        multispan.push_span_label(ident.span, "required by a bound in this");
-                    }
-                }
```

Comment:
@estebank I removed this code piece, (see 402714f4a7efa28ea12fa0db1864a0ca81990be4), causing all the removals. The information is redundant as there's always a follow up span that actually points to the bound.

Reply:
The bound is always pointed, yes, but that span was in order to let you scan the cli output and understand what item the bound was coming from, regardless of whether the bound is defined in the same line of as the ident or not. It feels like a regression to me, but maybe people hate it?